examples/lvn.py

```diff
@@ -205,6 +208,9 @@ def _fold(num2const, value):
             const_args = [num2const[n] for n in value.args]
             return FOLDABLE_OPS[value.op](*const_args)
         except KeyError:  # At least one argument is not a constant.
+            if value.op in {'eq', 'ne'} and value.args[0] == value.args[1]:
+                # Equivalent arguments may be evaluated for equality.
```

Comment:
Maybe worth including an example, because this can be a little surprising? That is, what we're looking for is stuff like `eq x x` where `x` need not be a constant—it can be any variable. The weird thing is that this is not _exactly_ a value numbering thing (it depends on variables, not values), although it is convenient to do here.

Reply:
Done. Wouldn't this fall under constant folding rather than LVN?

Reply:
Yeah, it's a little fuzzy. It's just that all the other constant folding we do *during* LVN is based on value numbers, not on variable identity. So this one breaks the mold.